src/server/scripts/Kalimdor/zone_desolace.cpp

```diff
@@ -88,12 +88,20 @@ class npc_aged_dying_ancient_kodo : public CreatureScript
                     DoCast(me, SPELL_KODO_KOMBO_DESPAWN_BUFF, true);
 
                     me->UpdateEntry(NPC_TAMED_KODO);
+                    me->CombatStop();
+                    me->DeleteThreatList();
+                    me->SetSpeed(MOVE_RUN, 0.6f, true);
                     me->GetMotionMaster()->MoveFollow(caster, PET_FOLLOW_DIST, me->GetFollowAngle());
+                    me->setActive(true);
```

Comment:
But this can have problems and the setactive won't go to false if the quest fails or did not finish. So the grid will remain active at all times so this can't be merged until that is fixed.